script.module.kodiswift/lib/kodiswift/plugin.py

```diff
@@ -0,0 +1,356 @@
+    """
+
+    def __init__(self, name=None, addon_id=None, plugin_file=None,
+                 info_type=None):
+        """
+        Args:
+            name (Optional[str]): The name of the plugin, e.g. 'Hello Kodi'.
+            addon_id (Optional[str): The Kodi addon ID for the plugin,
+                e.g. 'plugin.video.hellokodi'. This parameter is now optional
+                and is really only useful for testing purposes. If it is not
+                provided, the correct value will be parsed from the
+                addon.xml file.
+            plugin_file (Optional[str]): If provided, it should be the path
+                to the plugin.py file in the root of the addon directory.
+                This only has an effect when kodiswift is running on the
+                command line. Will default to the current working directory
+                since kodiswift requires execution in the root addon directory
+                anyway. The parameter still exists to ease testing.
+            info_type (Optional[str):
+        """
+        self._name = name
+        self._routes = []
+        self._view_functions = {}
+        self._addon = xbmcaddon.Addon()
+
+        self._addon_id = addon_id or self._addon.getAddonInfo('id')
+        self._name = name or self._addon.getAddonInfo('name')
+
+        self._info_type = info_type
+        if not self._info_type:
+            types = {
+                'video': 'video',
+                'audio': 'music',
+                'image': 'pictures',
+            }
+            self._info_type = types.get(self._addon_id.split('.')[1], 'video')
+
+        # Keeps track of the added list items
+        self._current_items = []
+
+        # Gets initialized when self.run() is called
+        self._request = None
+
+        # A flag to keep track of a call to xbmcplugin.endOfDirectory()
+        self._end_of_directory = False
+
+        # Keep track of the update_listing flag passed to
+        # xbmcplugin.endOfDirectory()
+        self._update_listing = False
+
+        # The plugin's named logger
+        self._log = setup_log(self._addon_id)
+
+        # The path to the storage directory for the addon
+        self._storage_path = xbmc.translatePath(
+            'special://profile/addon_data/%s/.storage/' % self._addon_id)
+        if not os.path.isdir(self._storage_path):
+            os.makedirs(self._storage_path)
+
+        # If we are running in CLI, we need to load the strings.xml manually
+        # Since kodiswift currently relies on execution from an addon's root
+        # directly, we can rely on cwd for now...
+        if kodiswift.CLI_MODE:
+            from kodiswift.mockxbmc import utils
+            if plugin_file:
+                plugin_dir = os.path.dirname(plugin_file)
+            else:
+                plugin_dir = os.getcwd()
+            strings_fn = os.path.join(
+                plugin_dir, 'resources', 'language', 'English', 'strings.po')
```

Comment:
```suggestion
                plugin_dir, 'resources', 'language', 'resources.language.en_gb', 'strings.po')
```

Reply:
I'll do the modification, but it is weird that the strings.po file as indicated above does not actually exist in the package.

Reply:
It doesn't because kodiswift (xbmcswift2) could be used without running Kodi (it's basically a framework to develop add-ons). So this would allow strings to be parsed if the actual add-on being developed have them (from the command line). The thing is, we don't allow these old paths anymore in addons...that being the reason for my request here.

Out of curiosity I ported xbmcswift2 to matrix to easy the effort of porting add-ons that depend on that to matrix (radio.de, NASA, vim casts,etc). I don't really want to develop or support it so in the end, I might just drop support for the framework in the near future. Legacy is a pain :)